ui/hooks/useTokensToSearch.js

```diff
@@ -57,7 +59,10 @@ export function getRenderableTokenData(
     (tokenData) => tokenData.address === address?.toLowerCase(),
   );
 
-  const usedIconUrl = iconUrl || tokenMetadata?.iconUrl || token?.image;
+  const tokenIconUrl = useTokenDetection
+    ? tokenMetadata?.iconUrl || iconUrl
+    : iconUrl || tokenMetadata?.iconUrl;
```

Comment:
Using terniary operator would make this code unclear and hard to read. Lets do something like:
```suggestion
  let tokenIconUrl = ''; 
  if(useTokenDetection){
    tokenIconUrl = tokenMetadata?.iconUrl || iconUrl
   }else{
   tokenIconUrl = iconUrl || tokenMetadata?.iconUrl;
   }
```

Reply:
Thank you for the review! I agree that it's best to avoid more complex logic with ternary operators. We actually have another [PR](https://github.com/MetaMask/metamask-extension/pull/15562) that will use the new token images service even if `useTokenDetection` is false, so I'm closing this one.